lib/hsa/mcwamp_hsa.cpp

```diff
@@ -708,7 +713,9 @@ struct HSAOpCoord
 // Base class for the other HSA ops:
 class HSAOp : public Kalmar::KalmarAsyncOp {
 public:
-    HSAOp(Kalmar::KalmarQueue *queue, hc::hcCommandKind commandKind) ;
+    typedef activity_prof::ActivityProf<HSAOpCoord> ActivityProf;
```

Comment:
prefer not creating a `typedef` for `ActivityProf`

Reply:
Because ActivityProf is template I'm accessing it from HSAOp class for static fields instantiating like the following:
template<> std::atomic<record_id_t> HSAOp::ActivityProf::_glob_record_id(0);

I can not pass op-coord and pass device_id and queue_id explicitly so I will be able to remove the template parameter and it will resolve the concerns about the template.

Reply:
I mean instead of passing op_coord reference I can pass explicitly device_id and queue_id